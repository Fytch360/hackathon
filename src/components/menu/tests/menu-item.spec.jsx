import React from 'react'
import { shallow } from 'enzyme'

import { MenuItem } from '..'

describe('<MenuItem />', () => {
    test('is available', () => {
        expect(MenuItem).toBeDefined()
    })

    test('Snapshot', () => {
        const wrapper = shallow(
            <div>
                <MenuItem title="testTitle" />
                <MenuItem icon="icon:core/cards/mc-36-mastercard-black-edition" />
                <MenuItem icon="icon:core/cards/mc-36-mastercard-black-edition" title="testTitle" />
            </div>
        )

        expect(wrapper).toMatchSnapshot()
    })

    test('Should render title', () => {
        const title = 'Exactly NOT MenuItem title'
        const wrapper = shallow(
            <MenuItem title={title} />
        )

        expect(wrapper.text()).toBe(title)
    })

    test('Should call onClick and onClose functions when MenuItem was clicked', () => {
        const onClickMock = jest.fn()
        const onCloseMock = jest.fn()
        const wrapper = shallow(
            <MenuItem title="testTitle" onClick={onClickMock} onClose={onCloseMock} />

        )

        expect(onClickMock.mock.calls.length).toBe(0)
        expect(onCloseMock.mock.calls.length).toBe(0)

        wrapper.find('button')
            .simulate('click', {})

        expect(onClickMock.mock.calls.length).toBe(1)
        expect(onCloseMock.mock.calls.length).toBe(1)
    })
})
