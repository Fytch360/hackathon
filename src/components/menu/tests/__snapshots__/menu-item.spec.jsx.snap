// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> Snapshot 1`] = `
<div>
  <MenuItem
    title="testTitle"
  />
  <MenuItem
    icon="icon:core/cards/mc-36-mastercard-black-edition"
  />
  <MenuItem
    icon="icon:core/cards/mc-36-mastercard-black-edition"
    title="testTitle"
  />
</div>
`;
